autowarefoundation/autoware
# [feature] ndt matching safe monitor

## Status
**PRODUCTION / DEVELOPMENT**

## Description
This node tries to solve autowarefoundation/autoware_ai#105. After an statistical analysis of NDT matching node in both failing and non-failing scenarios, criteria to decide when NDT fails was obtained. This node implements a simple, yet effective solution.

## Todos

## Steps to Test or Reproduce
1. Checkout this branch
1. Compile ./catkin_make_release
1. Launch rviz
1. Add an `OverlayText` display type to RVIZ and select `/ndt_monitor/ndt_info_text`
1. Play a rosbag with PCD map
1. Initialize NDT matching 
1. For even better results use GNSS if available for automatic reinitialization.
1. Launch this node `roslaunch ndt_localizer ndt_matching_monitor.launch`
1. Cause a fault in localization, i.e. increasing the `voxel_grid_filter` to an unreasonable value, and return it.

The localization should be halted if no GNSS is available or restarted if it is.

More information can be found in the README.md contained in the `ndt_matching_monitor`.

![image](https://user-images.githubusercontent.com/7009053/37969829-c5cc6774-320c-11e8-8cfd-313fb7dc5a07.png)
